packages/sources/coingecko/src/endpoint/crypto.ts

```diff
@@ -101,20 +101,43 @@ const handleBatchedRequest = (
 }
 
 export const execute: ExecuteWithConfig<Config> = async (request, context, config) => {
-  const validator = new Validator(request, inputParameters, {}, { overrides })
+  const validator = new Validator(request, inputParameters, {}, { overrides: internalOverrides })
 
   const endpoint = validator.validated.data.endpoint
   const jobRunID = validator.validated.id
-  const base = validator.overrideSymbol(AdapterName)
+  const base = validator.validated.data.base
   const quote = validator.validated.data.quote
   const coinid = validator.validated.data.coinid
-  let idToSymbol = {}
-  let ids = coinid
-  if (!ids) {
-    const coinIds = await getCoinIds(context, jobRunID)
-    const symbols = Array.isArray(base) ? base : [base]
-    idToSymbol = getSymbolsToIds(symbols, coinIds)
-    ids = Object.keys(idToSymbol).join(',')
+
+  let ids: string
+  let idToSymbol: OverrideToOriginalSymbol = {}
+  if (!coinid) {
+    let overrider: Overrider = {} as Overrider
+    try {
+      overrider = new Overrider(internalOverrides, request.data?.overrides, AdapterName)
+    } catch (untypedError) {
+      const error = untypedError as Error
+      new AdapterError({
+        jobRunID: validator.validated.id,
+        statusCode: 400,
+        message: error.message,
+        cause: error,
+      })
+    }
+    const [overriddenCoins, remainingSyms] = overrider.performOverrides(base)
+    let requestedCoins = overriddenCoins
+    if (remainingSyms.length > 0) {
+      const coinsResponse = await getCoinIds(context, jobRunID)
+      requestedCoins = Overrider.convertRemainingSymbolsToIds(
+        overriddenCoins,
+        remainingSyms.map((sym) => sym.toLowerCase()),
```

Comment:
It would be cleaner to handle the lowercasing within the Overrider

Reply:
I have to map toLowerCase or toUpperCase depending on how each DP formats their symbols, however I can make this better